packages/ui/src/components/va-virtual-scroller/useVirtualScrollerSizes.ts

```diff
@@ -0,0 +1,97 @@
+import { PropType, ExtractPropTypes, ref, Ref, shallowRef, computed, watch, onMounted } from 'vue'
+
+import { useEvent } from '../../composables'
+import { warn, isParsablePositiveMeasure } from '../../services/utils'
+
+const validateSizeProp = (v: number | string, propName: string) => {
+  const isProperValue = isParsablePositiveMeasure(v)
+
+  !isProperValue &&
+  warn(`[va-virtual-scroller] ${propName} should be number or parsable int greater or equal to 0. Provided: ${v}.`)
+
+  return isProperValue
+}
+
+export const useVirtualScrollerSizesProps = {
+  horizontal: { type: Boolean, default: false },
+  itemSize: {
+    type: [Number, String] as PropType<string | number>,
+    default: 0,
+    validator: (v: number | string) => { return validateSizeProp(v, 'itemSize') },
+  },
+  wrapperSize: {
+    type: [Number, String] as PropType<string | number>,
+    default: 100,
+    validator: (v: number | string) => { return validateSizeProp(v, 'wrapperSize') },
+  },
+}
+
+export const useVirtualScrollerSizes = (
+  props: ExtractPropTypes<typeof useVirtualScrollerSizesProps>,
+  scrollPosition: Ref<number>,
+) => {
+  const list = shallowRef<HTMLElement>()
+  const wrapper = shallowRef<HTMLElement>()
+
+  const parseSizeValue = (value: number | string) => {
+    if (typeof value === 'string') {
+      const parsedValue = parseInt(value)
```

Comment:
What would be if we get here `NaN` value? Is it possible?

Reply:
It's possible only if `value` contains no digits at all. This situation is regulated via props validator but i'll add one more check here just in case.